0030-simple-classical-representations.md

```diff
@@ -0,0 +1,260 @@
+# RFC Title
+
+| **Status**        | **Proposed/Accepted/Deprecated** |
+|:------------------|:---------------------------------------------|
+| **RFC #**         | ####                                         |
+| **Authors**       | Jake Lishman (jake.lishman@ibm.com)          |
+| **Deprecates**    | None                                         |
+| **Submitted**     | 2023-03-15                                   |
+| **Updated**       | 2023-03-15                                   |
+
+
+## Summary
+
+This proposes an experimental representation for operations on classical bits solely within conditions.
+This is not required to be the final design for classical handling in Terra, it is meant to be something simple that doesn't entangle existing Terra objects too closely, so can be deprecated in favour of a more complete system in the future.
+
+
+## Motivation
+
+- Current hardware is gaining support for runtime operations and subsequent conditions on classical bits
+- IBM dynamic-circuits hardware can do bit-level calculations, but there is no way to represent this in Qiskit to pass it on
+- On Terra, we need more of a feel for what will actually help us represent more complex classical constructs
+
+
+## User Benefit
+
+- Users of IBM dynamic circuits will be able to take advantage of the classical-processing capabilities of those compilers from within Qiskit
+- This support is not limited to IBM so may extend further, but we are not aware of other dynamic-circuits hardware accessible through Qiskit
+- For Terra development: this is a "build one to throw away"-type sytem, so we can see in actual usage what works and doesn't for us before we commit to further classical processing.
+
+
+## Design Proposal
+
+The intent of this RFC is to define an exploratory representation for a simple subset of classical bit-level operations, which expand the current state of Terra's `IfElseOp.condition`.
+This representation is not required to be the base for future expansion; it will be permissible to introduce a completely different classical-value type system if that is more appropriate, and deprecate this one.
+To that end, this proposal will not include adding any additional public methods to any existing Terra classes that might conflict with an alternative implementation.
+In particular, this proposal will not consider adding ease-of-use arithmetic methods to `Clbit` or `ClassicalRegister`, nor will it add expressions that can be used as the parameters to gates.
```

Comment:
I can add a couple of lines about that if you like, though there's equally no hard guarantee that it _will_ be removed; I was just designing such that it could be our standard methods of deprecation.

Reply:
Added a note about not requiring this to be removed and a short note on deprecations in e6dd506.